File: SerpentineApi/SerpentineApi/Responses/UserResponse.cs
using System.Text.Json.Serialization;
using SerpentineApi.Helpers;

namespace SerpentineApi.Responses;

public class UserResponse : BaseResponse
{
    public string FullName { get; set; } = "";
    public string Username { get; set; } = "";
    public string ProfilePictureUrl { get; set; } = "";
    public int Age { get; set; } = 0;

    public DateTime DayOfBirth { get; set; } = DateTime.Now;

    [JsonIgnore]
    public UserRoles Role { get; set; } = UserRoles.User;

    public int AccessLevel { get; set; } = 0;

}

File: serpentine-client/src/models/responses/user-response.ts
export class UserResponse {
    id: number = 0;
    createdAt: Date = new Date();
    updatedAt: Date = new Date();
    fullName: string = '';
    username: string = '';
    profilePictureUrl: string = '';
    age: number = 0;
  
    constructor(
      fullName: string = '',
      username: string = '',
      profilePictureUrl: string = '',
      age: number = 0
    ) {
      this.fullName = fullName;
      this.username = username;
      this.profilePictureUrl = profilePictureUrl;
      this.age = age;
    }
  }
  